src/scrapers/unionbank.js

```diff
@@ -0,0 +1,283 @@
+    ];
+}
+
+function getAmountData(amountStr) {
+  const amountStrCopy = amountStr.replace(',', '');
+  return parseFloat(amountStrCopy);
+}
+
+function getTxnAmount(txn) {
+    const credit = getAmountData(txn.credit).amount;
+    const debit = getAmountData(txn.debit).amount;
+    return (Number.isNaN(credit) ? 0 : credit) - (Number.isNaN(debit) ? 0 : debit);
+}
+
+function convertTransactions(txns) {
+    return txns.map((txn) => {
+        const convertedDate = moment(txn.date, DATE_FORMAT).toISOString();
+        const convertedAmount = getTxnAmount(txn);
+        return {
+            type: NORMAL_TXN_TYPE,
+            identifier: txn.reference ? parseInt(txn.reference, 10) : null,
+            date: convertedDate,
+            processedDate: convertedDate,
+            originalAmount: convertedAmount,
+            originalCurrency: SHEKEL_CURRENCY,
+            chargedAmount: convertedAmount,
+            status: txn.status,
+            description: txn.description,
+            memo: txn.memo,
+        };
+    });
+}
+
+function getTransactionDate(tds, txnDetailsObj) {
+    txnDetailsObj.date = tds[DATE_HEADER].innerText;
+}
+
+function getTransactionDescription(tds, txnDetailsObj) {
+    txnDetailsObj.description = tds[DESCRIPTION_HEADER].innerText;
+}
+
+function getTransactionReference(tds, txnDetailsObj) {
+    txnDetailsObj.reference = tds[REFERENCE_HEADER].innerText;
+}
+
+function getTransactionDebit(tds, txnDetailsObj) {
+    txnDetailsObj.debit = tds[DEBIT_HEADER].innerText;
+}
+
+function getTransactionCredit(tds, txnDetailsObj) {
+    txnDetailsObj.credit = tds[CREDIT_HEADER].innerText;
+}
+
+function extractTransactionDetails(txnRow, txnStatus) {
+    let txnDetailsObj = {status: txnStatus};
+    let tds = txnRow.innerTds;
+    getTransactionDate(tds, txnDetailsObj);
+    getTransactionDescription(tds, txnDetailsObj);
+    getTransactionReference(tds, txnDetailsObj);
+    getTransactionDebit(tds, txnDetailsObj);
+    getTransactionCredit(tds, txnDetailsObj);
+    return txnDetailsObj;
+}
+
+function isExpendedDescRow(txnRow) {
+    return txnRow.id !== undefined && txnRow.id !== null && txnRow.id === 'rowAdded';
+}
+
+function editLastTransactionDesc(txnRow, lastTxn) {
+    lastTxn.description = lastTxn.description + " " + txnRow.innerText;
+    return lastTxn;
+}
+
+function handleTransactionRow(txns, txnsTableHeaders, txnRow, txnType) {
+    if (isExpendedDescRow(txnRow)) {
+        txns.push(editLastTransactionDesc(txnRow, txns.pop()))
+    } else {
+        txns.push(extractTransactionDetails(txnRow, txnType));
+    }
+}
+
+async function extractTransactionsFromTable(page, tableTypeId, txnType) {
+    const txns = [];
+    if(transactionsTableHeaders === null) {
+        transactionsTableHeaders = await getTransactionsTableHeaders(page, tableTypeId);
+    }
+    const transactionsRows = await pageEvalAll(page, `#WorkSpaceBox #${tableTypeId} tr[class!='header']`, [], (trs) => {
+        return trs.map(async (tr) => ({
```

Comment:
@dratler I believe map isn't working nicely with async. in this case, since you need to call `await pageEvalAll` in each iteration, I would either convert this into a `for` loop, or use the concept of `Promise.all`.
see https://flaviocopes.com/javascript-async-await-array-map/